src/message-buffer.cpp

```diff
@@ -1,5 +1,76 @@
 #include "message-buffer.hpp"
 
+Message::Message(std::size_t maxSize)
+    : m_maxSize(maxSize)
+    , m_currSize(0)
+{
+    m_message.resize(m_maxSize);
+}
+
+template <typename T>
+void Message::save(const nlohmann::json& json)
+{
+    if (m_currSize + sizeof(T) > m_maxSize) {
+        return;
+    }
+    T value = json;
+    std::memcpy(m_message.data() + m_currSize, &value, sizeof(T));
+    m_currSize += sizeof(T);
```

Comment:
Não poderia inverter essa lógica evitando o if praticamente vazio?
```suggestion
    if (m_currSize + sizeof(T) <= m_maxSize) {
        T value = json;
        std::memcpy(m_message.data() + m_currSize, &value, sizeof(T));
        m_currSize += sizeof(T);
    }
    
```